docs/sharing.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Sharing
+type: docs
+menu: thanos
+slug: /Sharing.md
+---
+
+# Sharing
+
+Sharing is for Long Term Retention Storage.
+
+## Background
+
+Currently all components that read from object store assume that all the operations and functionality should be done based
+on **all** the available blocks that are present in the certain bucket's root directory.
+
+This is in most cases totally fine, however with time and allowance of storing blocks from multiple `Sources` into the same bucket,
+the number of objects in a bucket can grow drastically.
+
+This means that with time you might want to scale out certain components e.g:
+
+* Compactor: Larger number of objects does not matter much, however compactor has to scale (CPU, network) with number of Sources pushing blocks to the object storage.
+If you have multiple sources handled by the same compactor, with slower network and CPU you might not compact/downsample quick enough to cope with incoming blocks.
+    * This happens a lot if no compactor is deployed for longer periods and thus has to quickly catch up with large number of blocks (e.g couple of months).
+* Store Gateway: Queries against store gateway which are touching large number of Sources might be expensive, so it has to scale up with number of Sources if we assume those queries.
+    * Orthogonally we did not advertise any labels on Store Gateway's Info. This means that querier was not able to do any pre-filtering, so all store gateways in system are always touched for each query.
+
+### Reminder: What is a Source
+
+`Source` is a any component that creates new metrics in a form of Thanos TSDB blocks uploaded to the object storage. We differentiate Sources by `external labels`.
```

Comment:
This section can be removed. We need docs **how to use** sharding and **when to use** - nothing more.